// Copyright (c) 2021 The Pybind Development Team. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  https://www.apache.org/licenses/LICENSE-2.0

syntax = "proto2";

package pybind11.test;

import "pybind11_protobuf/tests/test.proto";

message BaseMessage {
  extensions 1000 to max
  [verification = UNVERIFIED];
}

extend BaseMessage {
  optional pybind11.test.IntMessage int_message = 1001;
}

message NestLevel1 {
  optional BaseMessage base_msg = 1;
}

message NestLevel2 {
  optional NestLevel1 nest_lvl1 = 1;
}

message AllowUnknownInner {
  extensions 2000 to max
  [verification = UNVERIFIED];
}

message AllowUnknownOuter {
  optional AllowUnknownInner inner = 1;
}
